internal/backend/config/set.go

```diff
@@ -0,0 +1,77 @@
+package config
+
+import (
+	"fmt"
+
+	"go.autokitteh.dev/autokitteh/sdk/sdkerrors"
+)
+
+type ComponentConfig interface {
```

Comment:
Rename to `Validated` or `Validator`

Reply:
the intention is to add more functionality in the future, so `Validator` is too narrow